src/services/CertificateGenerationService.ts

```diff
@@ -534,12 +534,13 @@ class CertificateGenerationService {
           let filteredTestResults: any[] = submittedTests.filter((testResult) => {
             const { testTypes } = testResult;
             if (!testTypes) {
-              return testResult;
-            }
-            // Only include passed and prs results (exclude failed and abandoned)
-            if (testTypes.filter((testType: ITestType) => testType.testResult === "pass" || testType.testResult === "prs").length) {
-              return testResult;
+              return false;
             }
+
+            return testTypes.some((testType: ITestType) =>
+              testType.testTypeClassification === "Annual With Certificate" &&
+                (testType.testResult === "pass" || testType.testResult === "prs")
+            );
           });
```

Comment:
With some simplification:
```suggestion
          const filteredTestResults = submittedTests.filter(({ testTypes }) => !testTypes
            ? testResult
            : testTypes.some((testType: ITestType) =>
                testType.testTypeClassification === "Annual With Certificate"
                && (testType.testResult === "pass" || testType.testResult === "prs"))
          );
```

Reply:
That worked nicely. Thank you.